packages/core-database/lib/wallet-manager.js

```diff
@@ -398,4 +451,25 @@ module.exports = class WalletManager {
     return wallet.balance.isZero() && !wallet.secondPublicKey && !wallet.multisignature && !wallet.username
   }
 
+  /**
+   * Update the vote balance of the delegate the vote is for.
+   * @param  {Object} sender
+   * @param  {Transaction} transaction
+   * @return {void}
+   */
+  __updateVoteBalance (sender, transaction) {
```

Comment:
that is quite complex operation.
- this is different if the tx is applied or reverted
- the votes change for sender AND receiver